casbin/enforcer.py

```diff
@@ -262,3 +262,17 @@ def get_permissions_for_user_in_domain(self, user, domain):
     def get_named_permissions_for_user_in_domain(self, ptype, user, domain):
         """gets permissions for a user or role with named policy inside domain."""
         return self.get_filtered_named_policy(ptype, 0, user, domain)
+
+    def get_all_roles_by_domain(self, domain):
+        """gets all roles associated with the domain.
+        note: Not applicable to Domains with inheritance relationship  (implicit roles)"""
+        g = self.model.model["g"]["g"]
+        policies = g.policy
+        roles = set()
+        for policy in policies:
+            if policy[len(policy) - 1] == domain:
+                role = policy[len(policy) - 2]
+                if role not in roles:
+                    roles.add(role)
+
+        return list(roles)
```

Comment:
@BustDot  The set is unordered. If the order is not required to get all roles here, then I think it is ok to do so.